pkg/xray/policies.go

```diff
@@ -469,9 +469,18 @@ func resourceXrayPolicyUpdate(ctx context.Context, d *schema.ResourceData, m int
 func resourceXrayPolicyDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
-	_, err := m.(*resty.Client).R().Delete("xray/api/v2/policies/" + d.Id())
-	if err != nil {
+	resp, err := m.(*resty.Client).R().Delete("xray/api/v2/policies/" + d.Id())
+	if err != nil && resp.StatusCode() == http.StatusInternalServerError {
+		d.SetId("")
 		return diag.FromErr(err)
 	}
 	return diags
 }
+
+func checkPolicy(id string, request *resty.Request) (*resty.Response, error) {
+	return request.Get("xray/api/v2/policies/" + id)
+}
+
+func testCheckPolicy(id string, request *resty.Request) (*resty.Response, error) {
```

Comment:
This should be in the test file, not this one.

Reply:
good catch

Reply:
moved to util_test.go